swarm/src/dial_opts.rs

```diff
@@ -283,6 +299,12 @@ impl WithoutPeerIdWithAddress {
         self.role_override = Endpoint::Listener;
         self
     }
+
+    pub fn override_port_mode(mut self, port_mode: PortMode) -> Self {
+        self.port_mode = port_mode;
+        self
+    }
```

Comment:
I think I'd prefer to not expose `PortMode` here and instead have two setters:

```rust
pub fn reuse_existing_port(&mut self) -> Self { }

pub fn allocate_new_port(&mut self) -> Self { }
```

Two benefits:

1. It is a forwards-compatible API which `DialOpts` is already designed for.
2. It avoids the user having to import `PortMode` when they use `DialOpts`.

Reply:
Sounds good to me.